underscore.js

```diff
@@ -1091,6 +1091,20 @@
     return obj;
   };
 
+  // Returns whether an object has a given set of `key:value` pairs.
+  _.matches = function(object, attrs) {
+    if (arguments.length < 2) return _.matcher(object);
+    var keys = _.keys(attrs), length = keys.length;
+    if (object == null) return !length;
+    var obj = Object(object);
```

Comment:
Note: I'm intentionally **not** assigning to `object` here to avoid an [arguments deopt](https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#31-reassigning-a-defined-parameter-while-also-mentioning-arguments-in-the-body-typical-example).